streams/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/rule/sql/SqlScript.java

```diff
@@ -141,23 +143,22 @@ private Values createValues(StreamlineEvent event) {
             }
             values.add(value);
         }
-        return values;
+        return CorrelatedEventsAwareValues.of(Collections.singletonList(event), values);
     }
 
-    private Collection<StreamlineEvent> convert(List<Values> result, final StreamlineEvent inputEvent) {
+    private Collection<StreamlineEvent> convert(List<CorrelatedEventsAwareValues> result,
+                                                final StreamlineEvent inputEvent) {
         Collection<StreamlineEvent> output = Collections.emptyList();
         if (result != null) {
             if (valuesConverter != null) {
-                output = Collections2.transform(result, new Function<Values, StreamlineEvent>() {
+                output = Collections2.transform(result, new Function<CorrelatedEventsAwareValues, StreamlineEvent>() {
                     @Override
-                    public StreamlineEvent apply(Values values) {
+                    public StreamlineEvent apply(CorrelatedEventsAwareValues values) {
                         return valuesConverter.convert(values, inputEvent);
                     }
                 });
-            } else if (result instanceof StreamlineEvent) {
```

Comment:
I couldn't imagine the case `List<Values>` is conform to `StreamlineEvent`, so I just removed.